titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java

```diff
@@ -47,4 +48,18 @@ private AgentFunctions() {
             return Optional.empty();
         }
     }
+
+    public static Function<AgentInstanceGroup, AgentInstanceGroup> withId(String newId) {
+        return ig -> ig.toBuilder().withId(newId).build();
+    }
+
+    public static Function<AgentInstanceGroup, AgentInstanceGroup> inState(InstanceGroupLifecycleState state) {
+        return ig -> ig.toBuilder().withLifecycleStatus(
+                InstanceGroupLifecycleStatus.newBuilder()
+                        .withState(state)
+                        .withTimestamp(System.currentTimeMillis())
+                        .withDetail("No reason given")
```

Comment:
Shouldn't this force a reason in the contract as there is generally always a reason to doing the transition.

Reply:
I created `AgentTestFunctions` and moved this method there.